.github/workflows/build.yml

```
@@ -42,8 +49,25 @@ jobs:
           p12-file-base64: ${{ secrets.CERTIFICATES_FILE_BASE64 }}
           p12-password: ${{ secrets.CERTIFICATES_PASSWORD }}
 
-      - name: Build and package
-        run: ./gradlew clean build --warning-mode all
+      - name: Run Tests (Linux)
+        if: runner.os == 'Linux'
+        run: |
+          export DISPLAY=:90
+          Xvfb -ac :90 -screen 0 1280x1024x24 > /dev/null 2>&1 &
+          ./gradlew -i test
+
+      - name: Run Tests (MacOS)
+        if: runner.os == 'macOS'
+        run: |
+          ./gradlew -i test
+
+      - name: Run Tests (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          ./gradlew -i test
+
+      - name: Build project
+        run: ./gradlew -i build -x test
```

The base plugin and java plugins ensure that
* build dependsOn check
* check dependsOn test

So I suggest reordering the steps to build and then test instead of the other way around

```suggestion
      - name: Build project
        run: ./gradlew -i build -x test

      - name: Run Tests (Linux)
        if: runner.os == 'Linux'
        run: |
          export DISPLAY=:90
          Xvfb -ac :90 -screen 0 1280x1024x24 > /dev/null 2>&1 &
          ./gradlew -i test

      - name: Run Tests (MacOS)
        if: runner.os == 'macOS'
        run: |
          ./gradlew -i test

      - name: Run Tests (Windows)
        if: runner.os == 'Windows'
        run: |
          ./gradlew -i test
```